nova/compute/api.py

```diff
@@ -4185,6 +4185,20 @@ def resize(self, context, instance, flavor_id=None, clean_shutdown=True,
         host_name can be set in the cold migration case only.
         """
 
+        # get all volumes attached to the instance and check if they are in a
+        # valid state for resize
+        bdms = objects.BlockDeviceMappingList.get_by_instance_uuid(
+                context, instance.uuid)
+        for bdm in bdms:
+            if bdm.volume_id:
+                volume = self.volume_api.get(context, bdm.volume_id)
+                if not ((volume['status'] == "in-use" and
+                        volume['attach_status'] == 'attached') or
+                        (volume['status'] == "available" and
+                        volume['attach_status'] == 'detached')):
+                    msg = _('Volume %s is not in a valid state for resize.')
+                    raise exception.InvalidVolume(msg % volume['id'])
```

Comment:
Looking at other places raising `InvalidVolume`, they all set `reason=msg`. Any reason not to do that here?